model/Check/System/AdvancedSearch/AdvancedSearchIndexationCheck.php

```diff
@@ -0,0 +1,167 @@
+<?php
+
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2022 (original work) Open Assessment Technologies SA.
+ */
+
+declare(strict_types=1);
+
+namespace oat\taoSystemStatus\model\Check\System\AdvancedSearch;
+
+use Throwable;
+use common_report_Report;
+use oat\tao\helpers\Template;
+use oat\oatbox\reporting\Report;
+use oat\tao\model\AdvancedSearch\AdvancedSearchChecker;
+use oat\taoAdvancedSearch\model\Index\Report\IndexSummarizer;
+
+class AdvancedSearchIndexationCheck extends AbstractAdvancedSearchCheck
+{
+    public const PARAM_ALLOWABLE_PERCENTAGE = 'allowablePercentage';
+    public const PARAM_BLACKLISTED_INDEXES = 'blacklistedIndexes';
+
+    /** @var float */
+    private $allowablePercentage;
+
+    /** @var array */
+    private $blacklistedIndexes;
+
+    /** @var Report[] */
+    private $reports = [];
+
+    /** @var Report[] */
+    private $errorReports = [];
+
+    public function __construct(array $params = [])
+    {
+        $this->allowablePercentage = (float) ($params[self::PARAM_ALLOWABLE_PERCENTAGE] ?? 80);
+        $this->blacklistedIndexes = $params[self::PARAM_BLACKLISTED_INDEXES] ?? [];
+
+        parent::__construct($params);
+    }
+
+    public function getDetails(): string
+    {
+        return __('Indexes population');
+    }
+
+    protected function getTemplate() : string
+    {
+        return Template::getTemplate('Reports/advancedSearchIndexesReport.tpl', 'taoSystemStatus');
+    }
+
+    protected function doCheck(): common_report_Report
+    {
+        $advancedSearchChecker = $this->getAdvancedSearchChecker();
```

Comment:
We might consider reseting the `reports` and `errorReports` here. If the service keeps in memory in a worker, we might  end up with several reports.

Reply:
Right now, we initialize these checkers manually: https://github.com/oat-sa/extension-tao-system-status/blob/master/model/CheckStorage/RdsCheckStorage.php#L139

Reply:
I had this issue here. It was initialized by DI container https://oat-sa.atlassian.net/browse/ADF-958

I would do it just to stay in the safe side. 

```suggestion
        $this->reports = [];
        $this->errorReports = [];
        
        $advancedSearchChecker = $this->getAdvancedSearchChecker();
```